docs: treatment of badges in the sidebar depending on development mode (#2233)
r-lib/pkgdown: man/rmd-fragments/sidebar-configuration.Rmd

-   `authors`: selected authors from the `DESCRIPTION`.

> -   `dev`: development status badges found in `README.md`/`index.md`. *This section is automatically removed for websites of released versions (see "Development mode" in `?build_site`. For all development modes, the badges are removed from the main content.*

Reviewer: Rather than saying when it's removed, might be clearer to state when it's shown?

Author: Is the new phrasing any better? See also #2240: if we add a parameter, then the docs will need to change again :zany_face:

> -   `dev`: development status badges found in `README.md`/`index.md`. *This section is automatically removed for websites of released versions (For all development modes, the badges are removed from the main content. See "Development mode" in `?build_site` for a summary of when badges are shown in the sidebar as opposed to not shown anywhere on the home page)*

Reviewer: ```suggestion
-   `dev`: development status badges extracted from `README.md`/`index.md`. This is only shown for "development" versions of websites; see "Development mode" in `?build_site` for details. 
```
